packages/core/src/utilities/derivedImageMetadataProvider.ts

```diff
@@ -0,0 +1,41 @@
+import { metaData } from '..';
+
+const state: Record<string, any> = {}; // Calibrated pixel spacing per imageId
+/**
+ * Simple metadata provider that stored the derived metadata in memory.
+ */
+const metadataProvider = {
```

Comment:
Could we call this something like registerTypeMetadataProvider?  It doesn't actually matter that it is for derived metadata, unless you plan on using that separately to allow clearing the state etc.  Having a single one that allows registering anything seems like it would be useful.

Reply:
GenericMetadataProvider